### ISSUE TITLE
[DOC] Missing documentation for search_pipelines in NodesInfo request section

### ISSUE THREAD

**Issue by lukas-vlcek** [opensearch-project/documentation-website#6544]
**What do you want to do?**
 
- [x] Request a change to existing documentation
- [ ] Add new documentation
- [ ] Report a technical problem with the documentation
- [ ] Other 

**Tell us about your request.** Provide a summary of the request and all versions that are affected.
 
There seems to be missing "metric" parameter documentation for `search_pipelines` in https://opensearch.org/docs/latest/api-reference/nodes-apis/nodes-info/

At least according to code there is such a metric in `NodesInfoRequest.Metric` enum and `public NodeInfo info()` in `NodeService` handles such parameter.

Cc: @msfroh 

**What other resources are available?** Provide links to related issues, POCs, steps for testing, etc.

n/a

**Comment by msfroh** [comment]
Yup.... this was a miss on my part when the search pipelines feature was added.

It was added to NodesInfo more so that nodes could coordinate amongst themselves (refusing to create a search pipeline unless all data nodes have the relevant processors installed). I forgot that NodesInfo is also user-visible.

**Comment by lukas-vlcek** [comment]
The thing is that with the [change being introduced here](https://github.com/opensearch-project/OpenSearch/pull/12497#issuecomment-1971425865) I also wanted to provide documentation update. Are you fine with me adding the missing `search_pipeline` parameter to NodesInfo doc and marking it as not included in the default set of "metrics"?

**Comment by msfroh** [comment]
That change isn't going to be backported to 2.x because it's a breaking change, right?

In that case, for now, the doc change should just be to add the parameter I think. 

Doc team, how are we planning to handle eventual 3.0 docs? I've kind of assumed that we're treating that as a "worry about it tomorrow problem" (but I like procrastinating).

**Comment by lukas-vlcek** [comment]
@msfroh There are two parts.
1) Missing `search_pipelines` parameter in the docs. When added it is relevant to 2.x as well, so it should be beckported if I understand correctly (I am also not sure what is the versioning story for docs).
2) Taking `search_pipelines` out of the default set of "metrics" (which will apply to some other "metrics" added in the future too, like `node_analyzers`). This will not go to 2.x.

That is why I talk about two PRs ... but again, I am not sure about the versioning strategy for docs at the moment.

**Comment by hdhalter** [comment]
Hi @lukas-vlcek , @msfroh , agreed, It sounds like two different PRs. We are updating the documentation for only the latest version of 2.x, which currently is 2.12. So, we can update the search_pipelines parameter there. can also create PRs for future (e.g., 3.0) documentation and just label it as 3.0 so we know not to merge it.
